src/joystick/windows/SDL_windows_gaming_input.c

```diff
@@ -444,6 +444,29 @@ WGI_JoystickInit(void)
         return SDL_SetError("RoInitialize() failed");
     }
 
+#ifndef __WINRT__
+    {
+        /* There seems to be a bug in Windows where a dependency of WGI can be unloaded from memory prior to WGI itself.
+         * This results in Windows_Gaming_Input!GameController::~GameController() invoking an unloaded DLL and crashing.
+         * As a workaround, we will keep a reference to the MTA to prevent COM from unloading DLLs later.
+         * See https://github.com/libsdl-org/SDL/issues/5552 for more details.
+         */
+        static PVOID cookie = NULL;
+        if (!cookie) {
+            typedef HRESULT (WINAPI *CoIncrementMTAUsage_t)(PVOID* pCookie);
+            CoIncrementMTAUsage_t CoIncrementMTAUsageFunc = (CoIncrementMTAUsage_t)WIN_LoadComBaseFunction("CoIncrementMTAUsage");
+            if (CoIncrementMTAUsageFunc) {
+                if (FAILED(CoIncrementMTAUsageFunc(&cookie))) {
+                    return SDL_SetError("CoIncrementMTAUsage() failed");
```

Comment:
Do we need to call WIN_RoUninitialize() in all the error cases in this function?

Reply:
`WGI_JoystickQuit()` will still call `WIN_RoUninitialize()` even if we fail initialization here.

However, it does look like there's a bug here if `WIN_RoInitialize()` itself fails. Calling `WIN_RoUninitialize()` in that case is wrong.

Reply:
Pushed 732595386146ff5c1246dc45448ec3c78cf3ca50 to fix that